tensorflow/core/kernels/histogram_op.cc

```diff
@@ -69,6 +72,79 @@ struct HistogramFixedWidthFunctor<CPUDevice, T, Tout> {
   }
 };
 
+#ifdef TENSORFLOW_USE_SYCL
+
+namespace {
+
+// Generate a matrix wich values are the index of the column.
+template <typename T>
+struct ColIndicesGenerator {
+  inline T operator()(const Eigen::array<Eigen::DenseIndex, 2>& idx) const {
+    return idx[1];
+  }
+};
+
+} //namespace
+
+template <typename T, typename Tout>
+struct HistogramFixedWidthFunctor<SYCLDevice, T, Tout> {
+  static Status Compute(OpKernelContext* context,
+                        const typename TTypes<T, 1>::ConstTensor& values,
+                        const typename TTypes<T, 1>::ConstTensor& value_range,
+                        int32 nbins, typename TTypes<Tout, 1>::Tensor& out) {
+    const SYCLDevice& d = context->eigen_device<SYCLDevice>();
+
+#ifdef TENSORFLOW_SYCL_NO_DOUBLE
+    using InternalT = T;
+#else
+    using InternalT = double;
+#endif  // TENSORFLOW_SYCL_NO_DOUBLE
+
+    auto values_size = values.size();
+    if (values_size == 0) {
+      out.device(d) = out.constant(0);
```

Comment:
Wouldn't `d.memset(out.data(), 0, size);` be faster here?
or even just return at this point?

Reply:
We can't return, out is not empty and needs to be set to 0.
Yeah memset could be faster, I think we would need "* sizeof( T)" though. Tbh I don't really like using it, it makes our specialization of the op unusable for CPUDevice. Because yes I still hope one day we wouldn't duplicate code as much!
I can update it if you really want

Reply:
oh OK I thought temporary is being `memset` to 0 when created.
And you are right.. lets not diverge from CPU code if we can!